File: backend/app/config/klien_supabase.py
"""Klien Supabase sisi server (service key — melewati RLS, hanya di backend)."""

from supabase import Client, create_client

from app.config.pengaturan import pengaturan

# Normalisasi: root .env memakai SUPABASE_URL berakhiran /rest/v1/,
# sedangkan supabase-py membutuhkan URL dasar proyek.
url = pengaturan.supabase_url
if "/rest/v1" in url:
    url = url.split("/rest/v1")[0].rstrip("/")

if not url or not pengaturan.supabase_key:
    raise RuntimeError("SUPABASE_URL / SUPABASE_KEY belum diisi di backend/.env")

klien: Client = create_client(url, pengaturan.supabase_key)

File: frontend/src/services/klien_supabase.js
import { createClient } from '@supabase/supabase-js';

// Klien Supabase (frontend) — memakai anon key + RLS, bukan service key.
const url = import.meta.env.VITE_SUPABASE_URL;
const kunciAnon = import.meta.env.VITE_SUPABASE_ANON_KEY;

if (!url || !kunciAnon) {
  throw new Error(
    'VITE_SUPABASE_URL / VITE_SUPABASE_ANON_KEY belum diisi di frontend/.env (salin dari .env root).',
  );
}

export const supabase = createClient(url, kunciAnon, {
  auth: {
    persistSession: true,
    autoRefreshToken: true,
    detectSessionInUrl: true,
  },
});